### PR TITLE
gh-90991: Include path to pyc file in pyc-related ImportErrors

### PR DESCRIPTION
**Pull request by hroncok**
https://bugs.python.org/issue46835

### CODE REVIEW

**Comment by encukou** [comment]
@brettcannon, as the importlib expert, does this make sense?

File: Lib/test/test_import/__init__.py [hunk 1]
```diff
@@ -1284,6 +1284,12 @@ def test_unencodable_filename(self):
         script_helper.assert_python_ok("-c", "mod = __import__(%a)" % name,
                                        __isolated=False)
 
+    def test_broken_pyc(self):
+        pyc = self.create_module("garbage", "...", ext=".pyc")
+        expected_msg = "bad magic number in {!r}: b'...'".format(os.path.abspath(pyc))
```
Comment (brettcannon) [inline Lib/test/test_import/__init__.py]:
```suggestion
        expected_msg = f"bad magic number in {os.path.abspath(pyc)!r}: b'...'"
```

**Comment by brettcannon** [comment]
I'm a bit torn on this one. On the one hand the exceptions being changed are rather low-level, so pointing _just_ at the file is probably fine and a bit helpful to find the bytecode file that is incorrect. But if the file path is really long, is that going to be annoying? And is anything lost by not having the module name?

**Comment by hroncok** [comment]
If I understand things correctly, the module name is a derivative of the path, isn't it?

**Comment by brettcannon** [comment]
> If I understand things correctly, the module name is a derivative of the path, isn't it?

Yes, but e.g. spotting `aggregate` in `/home/brettcannon/Repositories/brettcannon/stdlib-stats/__pycache__/aggregate.cpython-39.pyc` is not the easiest thing. But as I said earlier, maybe it doesn't matter since the module itself doesn't influence the exceptions being changed by this PR.

**Pull request closed without merging**